File: src/server/src/main/java/com/project/ports/rest/TagController.java
package com.project.ports.rest;

import com.project.application.models.tag.TagViewResource;
import com.project.application.services.TagService;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import java.util.List;

@Controller(value = "/tags")
public class TagController {
    private final TagService tagService;

    public TagController(TagService tagService) {
        this.tagService = tagService;
    }

    @Get
    public HttpResponse<List<TagViewResource>> getTags() {
        return HttpResponse.ok(this.tagService.getTags());
    }
}

File: src/server/src/main/kotlin/com/project/ports/rest/TagController.kt
package com.project.ports.rest

import com.project.application.models.tag.TagViewResource
import com.project.application.services.TagService
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller(value = "/tags")
open class TagController(private val tagService: TagService) {
    @Get
    open fun getTags(): HttpResponse<List<TagViewResource>> {
        return HttpResponse.ok(tagService.getTags())
    }
}
